src/Identifier/CallbackIdentifier.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SixtyEightPublishers\DoctrinePersistence\Identifier;
+
+use Nette;
+use SixtyEightPublishers;
+
+final class CallbackIdentifier implements IIdentifier
+{
+	use Nette\SmartObject;
+
+	/** @var callable  */
+	private $cb;
+
+	/**
+	 * @param callable $cb
+	 */
+	public function __construct(callable $cb)
+	{
+		$this->cb = $cb;
+	}
+
+	/**************** interface \SixtyEightPublishers\DoctrinePersistence\Identifier ****************/
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function getValue()
+	{
+		try {
+			$cb = $this->cb;
+			$value = $cb();
+		} catch (\Throwable $e) {
+		}
+
+		if (!isset($value)) {
```

Comment:
What about to use **early return** and get rid of nested if statement?

Reply:
hm.. I want to throw exception `MalformedIdentifierException` also if value returned from callback is `NULL` because Entity Identifier can't be `NULL`. This is why i didn't use **early return** in `try` block.

Reply:
I was thinking if the readability was improved by changes when I see them in the previous conversation and from this point of view the answer is yes 😉

Reply:
Thank you for improving my English 🙂